clippy_utils/src/consts.rs

```diff
@@ -567,6 +575,69 @@ impl<'a, 'tcx> ConstEvalLateContext<'a, 'tcx> {
     }
 }
 
+fn node_contains_cfg_expn(tcx: TyCtxt<'_>, n: Node<'_>) -> bool {
```

Comment:
Not sure where this should live. I can see a few options:
- here
- more generic version `clippy_utils::contains_expn_of(e: &Expr, name: &str)`
- in the `assertions_on_constants` lint

I can see it being useful in other lints like `needless_bool`.

Reply:
I agree with @giraffate The current impl would also no longer lint on things like `const _: bool = [cfg!("bla"); 10].len() > 5;`, just because there is a `cfg!` somewhere in the expression. But the `cfg!` in this case has absolutely no influence on the value of the resulting `bool`. 

I would also just check for direct `cfg!` and possibly extend this checks if other issues are opened about more complex cases that would make sense to allow.

Reply:
I figure a false negative is better than a false positive, but I understand not wanting to add so much complexity for little benefit.